### PR TITLE
Quick getChannel fix when returning Thread

### PR DESCRIPTION
**Pull request by xWildFire**
Causing error when e.g $message->getChannel is returning a thread

### CODE REVIEW

File: src/Discord/Discord.php [hunk 1]
```diff
@@ -1486,9 +1486,9 @@ public function __set(string $name, $value)
      *
      * @param string|int $channel_id Id of the channel.
      *
-     * @return Channel|null
+     * @return Channel|Thread|null
      */
-    public function getChannel($channel_id): ?Channel
+    public function getChannel($channel_id): Channel|Thread|null
```
Comment (davidcole1340) [inline src/Discord/Discord.php:1491]:
```suggestion
    public function getChannel($channel_id): ?Part
```

Union types were only introduced in PHP 8.0 but we need to support PHP 7.4.

**Review by davidcole1340 — CHANGES_REQUESTED** [review]
Just one small change 😃

File: src/Discord/Discord.php (hunk 1 above)
Comment (SQKo) [reply src/Discord/Discord.php:1491]: I think I know what he mean.. in API docs it is mentioned that [Get Channel](https://discord.com/developers/docs/resources/channel#get-channel) may include a thread member object if channel is a thread.

```suggestion
public function getChannel($channel_id): ?Part
{
    foreach ($this->guilds as $guild) {
        foreach ($guild->channels as $channel) {
            if ($channel->id == $channel_id) {
                return $channel;
            } else if ($thread = $channel->threads->get('id', $channel_id)) {
                return $thread;
            }
        }
    }

    if ($channel = $this->private_channels->get('id', $channel_id)) {
        return $channel;
    }

    return null;
}
```
Not sure if the nested loop is overkill

**Comment by SQKo** [comment]
I think I know what he mean.. in API docs it is mentioned that [Get Channel](https://discord.com/developers/docs/resources/channel#get-channel) may include a thread member object if channel is a thread.

```php
public function getChannel($channel_id): ?Part
{
    foreach ($this->guilds as $guild) {
        foreach ($guild->channels as $channel) {
            if ($channel->id == $channel_id) {
                return $channel;
            } else if ($thread = $channel->threads->get('id', $channel_id)) {
                return $thread;
            }
        }
    }

    if ($channel = $this->private_channels->get('id', $channel_id)) {
        return $channel;
    }

    return null;
}
```
Not sure if the nested loop is overkill

**Comment by davidcole1340** [comment]
Yeah I think it's best if `getChannel` doesn't try to check for threads. Could add another `getThread` function.

**Pull request closed without merging**